File: code/thesis_plots/software/_gmsh_meshing_algorithms.py
import gmsh

gmsh.initialize()
gmsh.fltk.initialize()

# Create domain
corners = [
    gmsh.model.geo.add_point(0, 0, 0),
    gmsh.model.geo.add_point(1, 0, 0),
    gmsh.model.geo.add_point(1, 1, 0),
    gmsh.model.geo.add_point(0, 1, 0)
]
domain_edge = [
    gmsh.model.geo.add_line(corners[i], corners[i+1]) for i in range(len(corners) - 1)
] + [gmsh.model.geo.add_line(corners[-1], corners[0])]

circumference = gmsh.model.geo.add_curve_loop(domain_edge)

surface = gmsh.model.geo.add_plane_surface([circumference])

gmsh.model.geo.synchronize()

algorithms = {
    1: "MeshAdapt",
    5: "Delaunay",
    6: "Frontal-Delaunay",
    7: "BAMG",
    8: "Frontal-Delaunay for Quads"
}

for i, alg in algorithms.items():
    gmsh.option.setNumber("Mesh.Algorithm", i)
    gmsh.model.mesh.generate(2)

    gmsh.write(f"plots/gmsh_meshing_algorithms_{alg}.png")

# Create a line in the mesh
start = gmsh.model.geo.add_point(0.2, 0.2, 0)
end = gmsh.model.geo.add_point(0.8, 0.8, 0)
line = gmsh.model.geo.add_line(start, end)
gmsh.model.geo.synchronize()

gmsh.model.mesh.embed(1, [line], 2, surface)

for i, alg in algorithms.items():
    gmsh.option.setNumber("Mesh.Algorithm", i)
    gmsh.model.mesh.generate(2)

    gmsh.write(f"plots/gmsh_meshing_algorithms_embedded_{alg}.png")


gmsh.finalize()

File: code/thesis_plots/software/gmsh_meshing_algorithms.m
resGridSize = 0.2;
domain = [1 1];

algorithms = [
    "meshadapt", "delaunay", "frontal", "bamg", "delquad"
];

for i = 1:length(algorithms)
    G = pebiGrid2DGmsh( ...
        resGridSize, ...
        domain, ...
        'meshAlgorithm', algorithms{i} ...
    );
    
    axis off;
    plotGrid(G, 'faceColor', 'none');
    exportgraphics(gcf,"plots/gmsh_meshing_algorithms_" + algorithms{i} + ".png");
    clf;
end

faceConstraints = {[0.2 0.2; 0.8 0.8]};
for i = 1:length(algorithms)
    G = pebiGrid2DGmsh( ...
        resGridSize, ...
        domain, ...
        'faceConstraints', faceConstraints, ...
        'meshAlgorithm', algorithms{i} ...
    );
    
    hold on;
    axis off;
    plotGrid(G, 'faceColor', 'none');
    plot(faceConstraints{1}(:, 1), faceConstraints{1}(:, 2), 'lineWidth', 2, 'color', 'magenta', 'LineStyle','-');
    exportgraphics(gcf,"plots/gmsh_meshing_algorithms_embedded_" + algorithms{i} + ".png");
    clf;
end